spyder/utils/introspection/manager.py

```diff
@@ -154,7 +183,18 @@ def _handle_timeout(self):
         self.waiting = False
         if self.pending:
             self._finalize(self.pending)
+        elif self.pending_request:
+            info = self.pending_request
+            self.pending_request = None
+            self.send_request(info)
         else:
+            self.timeout_counter += 1
+            if (self.timeout_counter > MAX_TIMEOUTS and
+                    not self.plugins_initialized['jedi']):

```

Comment:
Maybe the plugins would not have enough time to start and then it would restart them again and again, but if any of the plugin starts and gives a completion, the user gets the first option an then the `Tab` doesn't have any effect.